specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/openshiftclusters/preview/2024-08-12-preview/redhatopenshift.json

```diff
@@ -0,0 +1,2732 @@
+          }
+        },
+        "x-ms-examples": {
+          "Deletes a MachinePool with the specified subscription, resource group and resource name.": {
+            "$ref": "./examples/MachinePools_Delete.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "MachinePools"
+        ],
+        "summary": "Updates a MachinePool with the specified subscription, resource group and resource name.",
+        "description": "The operation returns properties of a MachinePool.",
+        "operationId": "MachinePools_Update",
+        "parameters": [
+          {
+            "$ref": "../../../../../../common-types/resource-management/v6/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v6/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v6/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "description": "The name of the OpenShift cluster resource.",
+            "required": true,
+            "type": "string",
+            "maxLength": 63,
+            "minLength": 1,
+            "pattern": "^[a-zA-Z0-9]$|^[a-zA-Z0-9][-_a-zA-Z0-9]*[a-zA-Z0-9]$"
+          },
+          {
+            "name": "childResourceName",
+            "in": "path",
+            "description": "The name of the MachinePool resource.",
+            "required": true,
+            "type": "string",
+            "maxLength": 63,
+            "minLength": 1,
+            "pattern": "^[a-zA-Z0-9]$|^[a-zA-Z0-9][-_a-zA-Z0-9]*[a-zA-Z0-9]$"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "The MachinePool resource.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/MachinePoolUpdate"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MachinePool"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.  If the resource doesn't exist, 404 (Not Found) is returned.  If any of the input parameters is wrong, 400 (Bad Request) is returned.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Updates a MachinePool with the specified subscription, resource group and resource name.": {
+            "$ref": "./examples/MachinePools_Update.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RedHatOpenShift/openshiftclusters/{resourceName}/secret/{childResourceName}": {
```

Comment:
>secret [](http://example.com/codeflow?start=141&length=6)

Secrets are not allowed in GET payloads. Must use list* POST action.

Reply:
see comment https://github.com/Azure/azure-rest-api-specs/pull/30004#discussion_r1887283477